src/runtime/virtcontainers/qemu.go

```diff
@@ -976,14 +977,14 @@ func (q *qemu) StopVM(ctx context.Context, waitOnly bool) error {
 	defer span.End()
 
 	q.Logger().Info("Stopping Sandbox")
-	if q.stopped {
+	if atomic.LoadInt32(&q.stopped) != 0 {
 		q.Logger().Info("Already stopped")
```

Comment:
This is not a really a bug regarding thread-safety. Instead of acquiring the mutex (q.mu) I used atomic to speed things a bit and to keep the logic the same as the one in clh. Atomic ops are required in clh, StopVM calls terminate isClhRunning while holding the mutex so trying to acquire it again in order to check stopped will trigger a deadlock. I'll move qemu bits to another commit.